CHANGELOG.md

```diff
@@ -60,7 +60,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 Initial release
 
-[Unreleased]: https://github.com/rust-embedded-community/serde-json-core/compare/v0.4.0...HEAD
```

Comment:
Add a new [Unreleased] tag and header above

Reply:
I wouldn't.
There is nothing unreleased and this is the changelog of the release. Adding top sections that refer to things that this release does not cover is confusing.
The first change that adds something after v0.5.0 should add the `Unreleased` section.

Reply:
This is a recommendation by [`keepachangelog`](https://keepachangelog.com/en/1.0.0/#effort). If we remove it, someone may just forget to add it in the future. If we keep it permanently there, it makes changing it into a new section exceptionally easy

Reply:
Hmm no. It says "At release time, you can move the Unreleased section changes into a new release version section.". And "move" as in "rename" is exactly what this is.

Forgetting to add the Unreleased section again is less likely as forgetting to add a changelog entry. You would hesitate to add an entry to an existing released section.

And I don't see how that makes anything easier. The new section has to be added in both cases. Same amount of work.
On the contrary it makes the actual release (the cargo package) much less useful and much more confusing as that now includes a changelog that has a section that is guaranteed to be out of date, inconsistent an meaningless: The link to the diff will be wrong and fill up while the section will by definition be empty.

Reply:
> And I don't see how that makes anything easier. 

The user only has to add a new bullet point under the `[Unreleased]` tag and it's clear where it's supposed to go for new contributors. It's not at all clear for them if they just see released changelogs and there's nowhere that would be clear for them to add new information.

> The new section has to be added in both cases. Same amount of work.

Correct. It's just easier to do it now to ensure it isn't forgotten.

> On the contrary it makes the actual release (the cargo package) much less useful and much more confusing as that now includes a changelog that has a section that is guaranteed to be out of date, inconsistent an meaningless:

I'm really not following what you mean here. crates.io doesn't use the CHANGELOG.md file at all. The only place I ever see it show up is if you manually view the source of the docs.rs entry.

For things like `dependabot` that actually parse the CHANGELOG, it is plenty smart enough to strip off the `[Unreleased]` portion and just grabs the section for the relevant releas, I believe.

>  The link to the diff will be wrong and fill up while the section will by definition be empty.

Huh? What link are you talking about? The link in the CHANGELOG is always correct because it's referring to diffs between two git tags

Reply:
> The user only has to add a new bullet point under the [Unreleased] tag 

That's an oversimplification. They would also have to figure out the prevalent style and sequence of Added/Changed/Removed sections and decide about the classification and add sections where if missing.

> The only place I ever see it show up is if you manually view the source of the docs.rs entry.
> Huh? What link are you talking about? The link in the CHANGELOG is always correct because it's referring to diffs between two git tags

If you want to look at a changelog, you'll head for github and on the right hand sid you'll quickly be guided to the release that you may be interested in. E.g.
look at the (current!) [v0.4.0 changelog](https://github.com/rust-embedded-community/serde-json-core/blob/v0.4.0/CHANGELOG.md
) and check whether that makes sense. The link content and the section content don't match at all. Specifically, for a release, why would you be interested to see things that are completely unrelated?
Sure, one of the first things after a release should be adding a new `Unreleased` section to the development branch.

Reply:
There is really no need for the release changelog to have an `Unreleased` section. What would be the point?

Reply:
That's a good point. When you navigate to that and click the `Unreleased` link, you indeed get a URL that changes in the future, which doesn't make sense for a release page. Thanks for clarifying :)